homeassistant/components/edl21/sensor.py

```diff
@@ -435,6 +434,12 @@ def __init__(self, electricity_id, obis, name, entity_description, telegram):
         }
         self._async_remove_dispatcher = None
         self.entity_description = entity_description
+        self._attr_name = name
```

Comment:
This matches existing behavior, but I think you should look at implementing the new entity naming conventions is a follow-up PR:
https://developers.home-assistant.io/blog/2022/07/10/entity_naming/